### PR TITLE
[Localization] Removing Test with Ignore File

### PR DESCRIPTION
**Pull request by tj-devel709**
I would like to remove this test and accompanying ignore file.
This test was useful when we manually had to add the translated error codes to the resx files.
Now that we have the OneLocBuild task handling this part for us, it is not needed.

It also causes issues when the OneLocBuild task gives us the translated resx files (such as [here](https://github.com/xamarin/xamarin-macios/pull/12220)) since this test fails and makes us go in and take out the translated error codes!

### CODE REVIEW

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs [hunk 1]
```diff
@@ -96,66 +96,6 @@ private string TranslateError (string culture, string errorCode)
 		IList<string> commonIgnoreList = null;
 		static string shortCommonPath = "xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationIgnore/common-Translations.ignore";
```
Comment (mandel-macaque) [inline tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs:97]: This does not look like is longer used, is it?

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs [hunk 2]
```diff
@@ -96,66 +96,6 @@ private string TranslateError (string culture, string errorCode)
...
-					Thread.CurrentThread.CurrentCulture = originalCulture;
-				}
-			}
-
-			Assert.IsEmpty (errorList.ToString (), $"The following errors were not translated. Add them to {shortCommonPath} or {shortCulturePath}");
-			Assert.IsEmpty (cultureIgnoreList.Except (cultureValidEntries), $"Not all error codes in {shortCulturePath} are valid or are repeated. Please remove.");
-			Assert.IsEmpty (commonIgnoreList.Except (commonValidEntries), $"Not all error codes in {shortCommonPath} are valid or are repeated. Please remove.");
-		}
-
 		IList<string> ReadFile (string path)
```
Comment (mandel-macaque) [inline tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs:99]: This does not look like it is longer used.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs (hunk 1 above)
Comment (tj-devel709) [reply tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs:97]: Yes, you are correct, thanks!

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs (hunk 2 above)
Comment (tj-devel709) [reply tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs:99]: Good call!

**Review by dalexsoto — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]
The merge issue needs to be fixed :)

**Comment by mandel-macaque** [comment]
Unrelated issues: https://github.com/xamarin/maccore/issues/2443

**Pull request merged**